src/main/java/net/lenni0451/classtransform/TransformerManager.java

```diff
@@ -264,13 +264,14 @@ public void hookInstrumentation(final Instrumentation instrumentation) throws Un
      */
     @Override
     public byte[] transform(ClassLoader loader, String className, Class<?> classBeingRedefined, ProtectionDomain protectionDomain, byte[] classfileBuffer) throws IllegalClassFormatException {
-        if (className == null) return classfileBuffer;
+        if (className == null) return null;
         try {
-            return transform(className.replace("/", "."), classfileBuffer);
+            byte[] newBytes = transform(className.replace("/", "."), classfileBuffer);
+            if (newBytes != classfileBuffer) return newBytes;
```

Comment:
I don't like that array comparison. If you use Arrays#equals I will merge this

Reply:
What do you have again ACMP_EQ :P